moveit_planners/chomp/chomp_motion_planner/src/chomp_planner.cpp

```diff
@@ -165,7 +188,33 @@ bool ChompPlanner::solve(const planning_scene::PlanningSceneConstPtr& planning_s
   }
   ROS_DEBUG_NAMED("chomp_planner", "Optimization took %f sec to create", (ros::WallTime::now() - create_time).toSec());
 
-  optimizer.optimize();
+  bool optimization_result = optimizer.optimize();
+
+  /// replan with updated parameters if no solution is found
+  if (params.getEnableFailureRecovery())
+  {
+    if (optimization_result == false && replan_cnt != params.getMaxRecoveryAttempts())
+    {
+      ROS_INFO("Re-planned with updated Chomp Parameters (learning_rate, ridge_factor, planning_time_limit, "
+               "max_iterations), attempt: # %d ",
+               (replan_cnt + 1));
+      ROS_INFO("Learning rate: %f ridge factor: %f planning time limit: %f max_iterations %d ", params.learning_rate_,
+               params.ridge_factor_, params.planning_time_limit_, params.max_iterations_);
+    }
+    if (optimization_result == false && replan_cnt < params.getMaxRecoveryAttempts())
+    {
+      replan_cnt++;
+      replan_flag = true;
+      goto RePlan;
```

Comment:
as @mamoll already pointed out, we cannot have ``goto`` in this code

Reply:
done, replaced with do while loop